[Julian] iP

Jnwkm opened the pull request:
# **DukePro**

>*"The essence of recursion and wishful thinking all begins in you."* - [Hartin Menz](https://www.comp.nus.edu.sg/images/resources/20210708_Source_Academy_Feature_YT.webp)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- *easy to learn*
- **~FAST~ SUPER FAST** to use

All you need to do is,

1. download it from [here](https://github.com/Jnwkm/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🍆

And it is **FREE!**

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Jnwkm committed:
Level-1

Jnwkm committed:
Level-2

Jnwkm committed:
Level-3

Jnwkm committed:
Level-4

Jnwkm committed:
A-TextUiTesting

Jnwkm committed:
Level-5

Jnwkm committed:
Level-6

Jnwkm committed:
Level-7

Jnwkm committed:
Level-8

Jnwkm committed:
Merge branch 'branch-Level-8'

Jnwkm committed:
A-MoreOOP

Jnwkm committed:
A-Packages

Jnwkm committed:
Merge branch 'add-gradle-support'

Jnwkm committed:
Revert "Merge branch 'add-gradle-support'"

This reverts commit 6889f7b968d4b0bb17765c792f161eaef91081d6, reversing
changes made to b463f1a673eeb5a9cc9f30e9ce70a366b9a10cb0.

Jnwkm committed:
Revert "Revert "Merge branch 'add-gradle-support'""

This reverts commit 93c2bc6cbe5135177bd313ff7e504e91477812cb.

Jnwkm committed:
A-JUnit

Jnwkm committed:
A-CodingStandard

Jnwkm committed:
Level-9

Jnwkm committed:
A-JavaDoc

Jnwkm committed:
Merge branch 'branch-A-CodingStandard'

Jnwkm committed:
Merge branch 'branch-Level-9'

weiquany commented on src/main/java/duke/TaskHandler.java:
```diff
@@ -0,0 +1,63 @@
+     *
+     * @param taskList TaskList to edit the list.
+     */
+
```
I think there is an extra blank line between the Javadoc comment and constructor that might not be in line with the CS2103 java coding standard.

weiquany commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,40 @@
+
+    /**
+     * Default constructor for Event.
+     * @param description Description of task.
```
Should there be an empty line between the description and parameter section?

weiquany commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,82 @@
+public class Storage {
+    private String filePath;
+
+    /**
+     * Default constructor for Storage given a filePath.
+     * @param filePath Filepath of where to load/save the data.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+
+    /**
+     * Reads the data into the list.
+     * @throws DukeException If the file is unreadable.
+     */
```
Should there be an empty line between the description and parameter section for the Javadoc comments?

weiquany commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,121 @@
+    public static ArrayList<Task> taskList = new ArrayList<>();
+
+    /**
+     * Copies contents of old list into new list.
```
Should there be an empty line between the description and parameter section?

weiquany commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,41 @@
+
+    /**
+     * Default constructor for Task.
+     * @param description The description of the task.
```
Should there be an empty line between the description and parameter section?

weiquany commented on src/main/java/duke/TaskHandler.java:
```diff
@@ -0,0 +1,63 @@
+     * Method to unmark an item in the list.
+     *
+     * @param input Input to be unmarked in the list.
+     */
```
I am not sure about this, but is it better to shorten the Javadoc comment description to "Deletes an item from the list."? Similarly for the other comments in some other parts of the application.

weiquany commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+ */
+public class Ui {
+
+    static void addedMsg(Task task) {
```
This may not be an issue, but perhaps Msg can be spelt out fully to Message. I learnt from an industry mentor that shortening variable/method names is not recommended (nor does it provide benefit) and could lead to misunderstandings/confusion if done in a large-scale project.

rylzxc replied:
I agree with the above comment.

weiquany commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,121 @@
+                Task removedTask = taskList.remove(index);
+                Ui.deleteMsg(removedTask);
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println(new DukeException("ERROR: there is no item in the list there!").getMessage());
```
This is a rather interesting way to print an error message. Do you care to explain the rationale? I am rather intrigued and wish to learn, thank you.

weiquany reviewed (commented):
Good job overall on adhering to the CS2103 coding standards! I had a difficult time looking for any possible mistakes, and may have made mistakes in my review, do feel free to correct me!

Jnwkm committed:
setup JavaFX

Jnwkm committed:
JavaFX GUI basics

Jnwkm committed:
Update README.md

Jnwkm committed:
Update README.md

rylzxc commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+/**
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
```
Hi, not exactly a coding violation but I was thinking to append 'Task' behind the class names for the types of Tasks?
```suggestion
public class DeadlineTask extends Task {
```
I would say this could be more useful for classes that potentially have multiple meanings -> Event could refer to button onClick event etc. People who view your code base can also easily navigate to whichever task type they want to review easily.

Let me know what you think! :)

rylzxc commented on src/main/java/duke/DukeDeadlineIncorrectException.java:
```diff
@@ -0,0 +1,7 @@
+package duke;
+
+public class DukeDeadlineIncorrectException extends DukeException {
```
I like that you specifically crafted exceptions for your program. I was thinking if these should be packaged into `exceptions` as they are all related.

Let me know!

rylzxc reviewed (commented):
Overall, good job on having clean code. Code was simple to understand too.

Jnwkm committed:
Level-10

Jnwkm committed:
Revert "Level-10"

This reverts commit f234f9682ca0659351693d07a191b15b79dbf062.

Jnwkm committed:
Level-10

Jnwkm committed:
bugfix

Jnwkm committed:
Revert "bugfix"

This reverts commit 89980899a067beb411b95d3968d41be45459f3de.

Jnwkm committed:
A-CheckStyle

emptygx commented on src/main/java/duke/EventTask.java:
```diff
@@ -0,0 +1,40 @@
+    private String formattedDateTime;
+    private String oldDate;
+
+    /**
+     * Default constructor for Event.
+     * @param description Description of task.
+     * @param dateTime Datetime of task.
+     */
```
Seems like there should be an empty line between description and parameter section. 🤔

emptygx commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+
+    /**
+     * Method to show contents of list.
+     * @return
```
Missing return description. 
```suggestion
     * @return Contents of list as a string.
```

emptygx commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,55 @@
+    static String addedMsg(Task task) {
+        return ("Got it. I've added this task: \n"
+                +
+                "   " + task.toString()
+                +
+                "\n Now you have " + TaskList.taskList.size()
```
Try to do line breaks before the "+" operator. 🥴

emptygx reviewed (commented):
LGTM overall, good job on following the coding standards, just some minor violations but nothing major! 👍

Jnwkm committed:
A-Assertions

Added assertions upon getting the index.

Documents important assumptions that should hold at various points in the code, such as the index cannot be <0.

Jnwkm committed:
A-CodeQuality

Jnwkm committed:
branch-A-CodeQuality

Jnwkm committed:
Merge pull request #1 from Jnwkm/branch-A-Assertions

A-Assertions

Jnwkm committed:
Merge pull request #2 from Jnwkm/branch-A-CodeQuality

branch-A-CodeQuality

Jnwkm committed:
C-BetterSearch

Find command: make matching case insensitive & matching with date.

Find command is case sensitive & restricted to description only.

A case insensitive find as well as matching with date is more user-friendly & convenient because users cannot be
expected to remember the exact case of the keywords.

Jnwkm committed:
Merge pull request #3 from Jnwkm/branch-A-CodeQuality

A-CodeQuality

Jnwkm committed:
Revert "branch-A-CodeQuality"

This reverts commit 42cbc26350e9d99ebe5d64a4451aff92618a0d4f.

Jnwkm committed:
Revert "Revert "branch-A-CodeQuality""

This reverts commit 72db6ed5c06f924ac4f7e7f1c4329235c76008bd.

Jnwkm committed:
Readme + Ui

Jnwkm committed:
bugfixes for Duke,Dialog,README

Jnwkm committed:
Update README.md

Jnwkm committed:
Update README.md

Jnwkm committed:
Update README.md

Jnwkm committed:
Update README.md

Jnwkm committed:
update saving functionality & ui fixes

Jnwkm committed:
Update Ui.png
